installer/ansible/roles/dashboard-installer/scenarios/source-code.yml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2018 The OpenSDS Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: check for opensds-dashboard source code existed
+  stat:
+    path: "{{ dashboard_work_dir }}"
+  register: opensdsexisted
+
+- name: download opensds-dashboard source code if not exists
+  git:
+    repo: "{{ dashboard_remote_url }}"
+    dest: "{{ dashboard_work_dir }}"
+    version: "{{ dashboard_repo_branch }}"
+  when:
+    - opensdsexisted.stat.exists is undefined or opensdsexisted.stat.exists == false
+
+- name: build and configure opensds dashboard
+  shell: "{{ item }}"
+  with_items:
+    - service apache2 stop
```

Comment:
If we do this way, will it not affect the dashboard of any other existing instance of the dashboard?!

Reply:
Done